classes/tdx-tezi-data-partition.bbclass

```diff
@@ -9,8 +9,15 @@ TDX_TEZI_DATA_PARTITION_TYPE ?= "ext4"
 # data partition label
 TDX_TEZI_DATA_PARTITION_LABEL ?= "DATA"
 
-# automount data partition at boot time
-TDX_TEZI_DATA_PARTITION_AUTOMOUNT ?= "1"
+# automount data partition at boot time; possible values:
+#  "-1": do not change fstab
+#   "0": add entry to fstab with the noauto option
+#   "1": add entry to fstab with the auto option
+TDX_TEZI_DATA_PARTITION_AUTOMOUNT ?= "${TDX_TEZI_DATA_PARTITION_AUTOMOUNT_DEFAULT}"
+
+# default value of above variable depends on tdx-encrypted being in use
+TDX_TEZI_DATA_PARTITION_AUTOMOUNT_DEFAULT = "1"
+TDX_TEZI_DATA_PARTITION_AUTOMOUNT_DEFAULT:tdx-encrypted = "-1"
```

Comment:
If a user looks at this, she might get confused about which variable needs to be set: `TDX_TEZI_DATA_PARTITION_AUTOMOUNT` or `TDX_TEZI_DATA_PARTITION_AUTOMOUNT_DEFAULT`, although `TDX_TEZI_DATA_PARTITION_AUTOMOUNT` is the one documented.

Makes sense to add a comment to emphasize that `TDX_TEZI_DATA_PARTITION_AUTOMOUNT_DEFAULT` is for internal usage, and the user should set the `TDX_TEZI_DATA_PARTITION_AUTOMOUNT` variable?

Reply:
Yeah, it does make sense.

Reply:
Done